packages/tx-construction/src/tx-builder/initializeTx.ts

```diff
@@ -35,36 +35,43 @@ export const initializeTx = async (
       changeAddressResolver: new StaticChangeAddressResolver(async () => addresses)
     });
 
-  const validityInterval = ensureValidityInterval(tip.slot, genesisParameters, props.options?.validityInterval);
-  const withdrawals: Cardano.Withdrawal[] = rewardAccounts
-    .map(({ rewardBalance: quantity, address: stakeAddress }) => ({
-      quantity,
-      stakeAddress
-    }))
-    .filter(({ quantity }) => !!quantity);
+  const customizeCb = props.customizeCb ? props.customizeCb : (tx: Readonly<TxBodyPreInputSelection>) => tx;
+
+  // Create transaction body that can be customized by the user via the customizeCb
+  const { body: b, auxiliaryData } = createPreInputSelectionTxBody({
+    auxiliaryData: props.auxiliaryData,
+    certificates: props.certificates,
+    collaterals: props.collaterals,
+    mint: props.mint,
+    outputs: [...(props.outputs || [])],
+    requiredExtraSignatures: props.requiredExtraSignatures,
+    scriptIntegrityHash: props.scriptIntegrityHash,
+    validityInterval: ensureValidityInterval(tip.slot, genesisParameters, props.options?.validityInterval),
+    withdrawals: rewardAccounts
+      .map(({ rewardBalance: quantity, address: stakeAddress }) => ({
+        quantity,
+        stakeAddress
+      }))
+      .filter(({ quantity }) => !!quantity)
+  });
+
+  const bodyPreInputSelection = customizeCb(b);
```

Comment:
`customizeCb` seems to be used only here: I would remove its declaration (a dozen of lines above) and...
```suggestion
  const bodyPreInputSelection = props.customizeCb ? props.customizeCb(b) : b;
```

Reply:
Done here: [ec76f62](https://github.com/input-output-hk/cardano-js-sdk/pull/1099/commits/ec76f62e8dff3be6334bd8bac694f380c9061544)